### PR TITLE
feat(flutter): update screenshots and session replay with masking options

### PR DESCRIPTION
**Pull request by martinhaintz**
## DESCRIBE YOUR PR
In Flutter we introduced redaction via masking for `Screenshots` and `Session Replay`, therefore I updated the docs.

There were already infos, for masking in the `Session Replay` which I consolidated and extended and linked to from `Screenshots` and `Session Replay`.

## IS YOUR CHANGE URGENT?  

Help us prioritize incoming PRs by letting us know when the change needs to go live.
- [ ] Urgent deadline (GA date, etc.): 
- [ ] Other deadline: 
- [x] None: Not urgent, can wait up to 1 week+

## PRE-MERGE CHECKLIST

*Make sure you've checked the following before merging your changes:*

- [x] Checked Vercel preview for correctness, including links
- [ ] PR was reviewed and approved by any necessary SMEs (subject matter experts)
- [ ] PR was reviewed and approved by a member of the [Sentry docs team](https://github.com/orgs/getsentry/teams/docs)

## EXTRA RESOURCES

- [Sentry Docs contributor guide](https://docs.sentry.io/contributing/)

### CODE REVIEW

File: platform-includes/enriching-events/attach-screenshots/flutter.mdx [hunk 1]
```diff
@@ -13,6 +13,8 @@ await SentryFlutter.init(
 );
 ```
 
+<PlatformContent includePath="replay/privacy-configuration" />
+
```
Comment (vaind) [inline platform-includes/enriching-events/attach-screenshots/flutter.mdx:17]: would it work to move the privacy header out of the shared page and add some intro specific to screenshots? i.e. that masking is not enabled by default unless you specify the privacy options.

Comment (martinhaintz) [reply platform-includes/enriching-events/attach-screenshots/flutter.mdx:26]: I added a specific intro to Screenshots and Session Replay. I hope this is, what you had in mind.

File: docs/platforms/flutter/session-replay/index.mdx [hunk 2]
```diff
@@ -57,50 +57,17 @@ Sampling allows you to control how much of your website's traffic will result in
...
 
-## Privacy
+## Redact Session Replay via `masking`
 
-The SDK is recording and aggressively redacting (masking) all `Text`, `EditableText`, and `Image` widgets.
-Masking in the Sentry Flutter SDK is based on Widget *types*, e.g. `Image`, not the string representation of the type (i.e. we check whether
-a `widgetInstance` should be masked by checking `if (widgetInstance is Image)` instead of `if (widgetInstance.runtimeType == 'Image')`).
-This means we can ensure masking works regardless of obfuscation in release builds and also works for subclasses.
-However, it also means we can only automatically mask widgets that are part of the Flutter SDK itself.
+By default, the SDK is recording and aggressively redacting (masking) all `Text`, `EditableText`, and `Image` widgets for <PlatformLink to="/session-replay/">`Session Replay`</PlatformLink>. To modify or disable this beahvior, use the `options.experimental.privacy` parameter.
```
Comment (vaind) [inline docs/platforms/flutter/session-replay/index.mdx:62]:
```suggestion
By default, the SDK is recording and aggressively redacting (masking) all `Text`, `EditableText`, and `Image` widgets for <PlatformLink to="/session-replay/">`Session Replay`</PlatformLink>. To modify or disable this behavior, use the `options.experimental.privacy` parameter.
```

**Review by vaind — APPROVED** [review]
Looks very good, thanks!

**Pull request merged**